tests/system/python/api/test_plugin_discovery.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test plugin discovery (north, south, filter, notificationDelivery, notificationRule) REST API """
+
+import subprocess
+import http.client
+import json
+from collections import Counter
+import pytest
+
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+def install_plugin(type, plugin, branch="develop", plugin_lang="python", use_pip_cache=True):
+    if plugin_lang == "python":
+        path = "$FOGLAMP_ROOT/tests/system/python/scripts/install_python_plugin {} {} {} {}".format(
+            branch, type, plugin, use_pip_cache)
+    else:
+        path = "$FOGLAMP_ROOT/tests/system/python/scripts/install_c_plugin {} {} {}".format(
+            branch, type, plugin)
+    try:
+        subprocess.run([path], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "{} plugin installation failed".format(plugin)
+
+    # Cleanup code
+    subprocess.run(["rm -rf /tmp/foglamp-{}-{}".format(type, plugin)], shell=True, check=True)
+    # TODO: FOGL-2642 - Remove notificationDelivery And notificationRule plugins
+
+
+@pytest.fixture
+def reset_plugins():
+    subprocess.run(["rm -rf $FOGLAMP_ROOT/plugins/south"], shell=True, check=True)
+    subprocess.run(["path=$FOGLAMP_ROOT/python/foglamp/plugins/south; "
+                    "find $path -maxdepth 1 | grep -v \"^${path}$\" | "
+                    "egrep -v '(__init__.py)' | xargs rm -rf"], shell=True, check=True)
+    subprocess.run(["path=$FOGLAMP_ROOT/python/foglamp/plugins/north; "
+                    "find $path -maxdepth 1 | grep -v \"^${path}$\" | "
+                    "egrep -v '(pi_server|ocs|common|empty|README.rst|__init__.py)' | xargs rm -rf"],
+                   shell=True, check=True)
+    subprocess.run(["path=$FOGLAMP_ROOT/plugins/north; "
+                    "find $path -maxdepth 1 | grep -v \"^${path}$\" | "
+                    "egrep -v '(PI_Server_V2|ocs_V2)' | xargs rm -rf"],
+                   shell=True, check=True)
+    subprocess.run(["path=$FOGLAMP_ROOT/plugins/filter; "
```

Comment:
let's have a script for this, and pass arg for south, north... if no arg passed, rm all